analytics2/utils/block_stats.py

```diff
@@ -198,6 +199,73 @@ def add_block_pop_density(bldg_pop: gpd.GeoDataFrame,
     return bldg_pop  
 
 
+def add_block_gini(bldg_pop: gpd.GeoDataFrame,
+                   values_col: str,
+                   block: Union[gpd.GeoDataFrame, str] = None,
+                   weights: libpysal.weights.weights.W = None
+                   ) -> gpd.GeoDataFrame:
+    """
+    Calculates the gini coefficient for the passed in values_col based on the passed in gpd.GeoDataFrame.
+    """
+    if 'block_id' not in bldg_pop.columns:
+        block = flex_load(block)
+        bldg_pop = add_block_id(bldg_pop, block)
+
+    if values_col not in bldg_pop.columns:
+        raise IndexError(f'{values_col} was not found in bldg_pop')
+
+    if weights is None:
+        weights_dict = {bldg_pop.loc[index_val]['bldg_id']: bldg_pop[bldg_pop['block_id'] == bldg_pop.loc[index_val]['block_id']]['bldg_id'].tolist() for index_val in bldg_pop.index.tolist()}
+        weights = W(weights_dict, ids=bldg_pop['bldg_id'].tolist())
+
+    bldg_pop['gini_'+values_col] = Gini(bldg_pop, values=values_col, spatial_weights=weights, unique_id='bldg_id').series
+
+    return bldg_pop
+
+
+def add_block_rectangularity(bldg_pop: gpd.GeoDataFrame,
+                             block: Union[gpd.GeoDataFrame, str] = None,
+                             ) -> gpd.GeoDataFrame:
+    """
+    Adds rectangularity for each block in the data frame and adds that to the bldg_pop geodf.
+    Calculated as (area)/(minimum bounding rotated rectangle area)
+    See: http://docs.momepy.org/en/stable/generated/momepy.Rectangularity.html#momepy.Rectangularity
+    """
+    if 'block_id' not in bldg_pop.columns:
+        block = flex_load(block)
+        bldg_pop = add_block_id(bldg_pop, block)
+
+    if 'block_area' not in bldg_pop.columns:
+        bldg_pop = add_block_area(bldg_pop, block)
+
+    bldg_pop['rectangularity'] = Rectangularity(bldg_pop, areas='block_area').series
+
+    return bldg_pop
+
+
+def add_building_adjacency(bldg_pop: gpd.GeoDataFrame, 
```

Comment:
Let's add a better description here of what this captures. I think it's fine to have a link to a paper but we should have some clear idea of what the measure is capturing and that can only be found (in my experience) by reading the source literature itself - how are the original authors using the metric? Does their methodology pertain to urban spaces in developing countries/slums? 

As we move forward with the analytics project, we have general goals (i.e. characterize urban space/slums in developing countries) but the specifics are open-ended. A good way to get a sense of what questions we can/cannot ask/answer is to become familiar with the literature. The source papers would be good starting points and I bet there's a lot of good shit in these papers. 

For example, in this paper they describe their methodological process as: "The  method to  develop a  configuration typology  consists of  following  steps: first,  defining the aggregation unit, second, selecting a set of relevant spatial properties, third, calculating the properties per aggregation unit, fourth, organising the information per aggregation unit in a geo-database and fifth, distinguishing the configuration types by using a non parametric statistical clustering method." <--This is precisely the process we'd like to apply for our analysis. Their paper is focusing on Amsterdam but the degree to which their analysis applies or doesnt' apply to cities of our choosing could be an interesting thing to explore.

That's sort of rambling but hopefully it's useful